src/persistence/index.js

```diff
@@ -76,6 +76,105 @@ export class PersistDB {
+     */
+    static toJS() {
+        return PersistDB.open().then(async db => {
+            const tx = db.database.transaction(
+                ['states', 'changes', 'contents'], 'readonly')
+
+            const [states, changes, contents] = await Promise.all([
+                tx.objectStore('states').getAll(),
+                tx.objectStore('changes').index('document').getAll(),
+                tx.objectStore('contents').getAll(),
+            ].map(promisify))
+
+            return {
+                name: DB_NAME,
+                version: DB_VERSION,
+                states,
+                changes,
+                contents,
+            }
+        }).catch(e => {
+            throw new Error(e)
+        })
+    }
+
+    /**
+     * Replace database from JS Object. This will delete existing database and replace
+     * ot with new content.
+     *
+     * If database does not exists, it will be created.
+     *
+     * If you provide @param ids then this will remove only data for those ids.
+     *
+     * @param {Object} newDB
+     * @param {string[]} ids
+     */
+    static replaceDatabase(newDB = {}, ids = []) {
+        const {
+            name: newName,
+            version: newVersion,
+            states: newStates = [],
+            changes: newChanges = [],
+            contents: newContents = [],
+        } = newDB
+
+        if (newName !== DB_NAME) {
+            throw new Error(`Provided name "${newName}" does not match database name "${DB_NAME}".`)
+        }
+        if (Number(newVersion) !== DB_VERSION) {
+            throw new Error(`Provided version "${newVersion}" does not match database version "${DB_VERSION}."`)
+        }
+
+        PersistDB.open().then(async db => {
+            const tx = db.database.transaction(
+                ['states', 'changes', 'contents'], 'readwrite')
+            const states = tx.objectStore('states')
+            const changes = tx.objectStore('changes')
+            const contents = tx.objectStore('contents')
+
+            if (ids.length) {
+                let promises = []
+                ids.map(id => {
+                    promises.push(promisify(states.delete(id)))
+                    promises.push(promisify(contents.delete(id)))
+                })
+                await Promise.all([
+                    ...promises,
+                    iterate(changes, async (cursor, value) => {
+                        if (ids.includes(value.document)) {
+                            cursor.delete()
+                        }
+                        cursor.continue()
+                    }),
+                ])
+            } else {
+                await Promise.all([
+                    states.clear(),
+                    changes.clear(),
+                    contents.clear(),
+                ].map(promisify))
+            }
+
+            await Promise.all([newStates.map(st => {
+                return promisify(states.put(st))
+            })])
```

Comment:
`Promise.all` accepts an array of promises, but here you pass it an one-element array containing an array of promises.